src/shogun/mathematics/linalg/ratapprox/logdet/LogDetEstimator.cpp

```diff
@@ -233,60 +219,78 @@ SGMatrix<float64_t> CLogDetEstimator::sample_without_averaging(
 	// call the precompute of the sampler
 	m_trace_sampler->precompute();
 
-	// for storing the aggregators that submit_jobs return
-	CDynamicObjectArray aggregators;
-	index_t num_trace_samples=m_trace_sampler->get_num_samples();
+	// for storing the result
```

Comment:
there is so much overlapping code here. Could we maybe refactor this to avoid that?
@lambday what are your thoughts here?